docs/reference/how-to/general.asciidoc

```diff
@@ -12,6 +12,34 @@ documents that match a particular query. If you need to do this, make sure to
 use the <<search-request-scroll,Scroll>> API.
 
 [float]
+[[maximum-document-size]]
+=== Avoid large documents
+
+Given that the default <<modules-http,`http.max_context_length`>> is set to
+100MB, Elasticsearch will refuse to index any document that is larger that
+that. You might decide to increase that particular setting, but Lucene still
+has a limit at about 2GB.
+
+But even without considering hard limits, large documents are usually not
+practical. Large documents put more stress on network, disk and on memory usage
+since their content will need to be retrieved by the `_search` API to build
+the response. Inverting this document can use an amount of memory that is a
+multiplier of the original size of the document. Proximity search (phrase
+queries for instance) and <<search-request-highlighting,highlighting>> also
+become more expensive since their cost directly depends on the size of the
```

Comment:
Hitting the large document in the response is going to be slow in ES as well because we always go to stored fields for the id. Unless you turn off storing _source. I'm not sure if that needs to be in the list, but it feels important because as it reads now I'd think "well, I just have to avoid phrase queries and highlighting".